src/__tests__/uiUtils.tes.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+  showStar,
+  discountCalculation,
+  getErrorMessageFromCode,
+} from "@/utils/uiUtils";
+import "@testing-library/jest-dom";
+
+describe("uiUtils", () => {
+  test("showStar returns correct value", () => {
+    expect(showStar(1)).toEqual("★☆☆☆☆");
+    expect(showStar(2)).toEqual("★★☆☆☆");
+    expect(showStar(1.5)).toEqual("★⯨☆☆☆");
```

Comment:
We would not want to see `★⯨☆☆☆` in the UI. Make sure that the function showStar never returns a broken UI. E.g. return <></> or null when unexpected value is passed to it.